test/ffmpeg-vaapi/encode/encoder.py

```diff
@@ -212,10 +216,15 @@ def check_output(self):
 
     # ipb mode
     ipbmode = 0 if vars(self).get("gop", 0) <= 1 else 1 if vars(self).get("bframes", 0) < 1 else 2
+    if vars(self).get("lowdelayb", None) is not None:
+      assert type(self.lowdelayb) is int and self.lowdelayb >= 0 and self.lowdelayb <= 2, "invalid lowdelayb value"
```

Comment:
Shouldn't this check be done way before attempting to run the pipeline command?

Reply:
e.g. right after the call to validate_caps() in encode()

Reply:
yes, updated based on your suggestion